src/components/SampleList/SampleList.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useMemo } from "react";
+import {
+  IonButton,
+  IonChip,
+  IonCol,
+  IonGrid,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonListHeader,
+  IonNote,
+  IonRow,
+  IonSearchbar,
+  SearchbarInputEventDetail,
+} from "@ionic/react";
+import { paths } from "../../Router";
+import { search as searchIcon } from "ionicons/icons";
+import NoneOr from "../NoneOr/NoneOr";
+import { SubmissionMetadata } from "../../api";
+import { IonSearchbarCustomEvent } from "@ionic/core/dist/types/components";
+import { useMiniSearch } from "react-minisearch";
+import { getSubmissionSamples } from "../../utils";
+
+interface SampleListProps {
+  submission: SubmissionMetadata;
+  collapsedSize?: number;
+}
+
+const SampleList: React.FC<SampleListProps> = ({
+  submission,
+  collapsedSize = 5,
+}) => {
+  const searchElement = React.useRef<HTMLIonSearchbarElement>(null);
+
+  const [collapsed, setCollapsed] = React.useState(true);
+
+  const [showSearch, setShowSearch] = React.useState(false);
+  const [searchTerm, setSearchTerm] = React.useState<string | null>();
+
+  const samples = useMemo(
+    // This is reversed so that the most recent samples are shown first
+    () => getSubmissionSamples(submission).toReversed(),
+    [submission],
+  );
+
+  const { search, searchResults } = useMiniSearch(samples, {
+    fields: ["samp_name"],
+    idField: "_index",
+    searchOptions: {
+      prefix: true,
+      combineWith: "AND",
+    },
+  });
+
+  const handleSearchInput = (
+    event: IonSearchbarCustomEvent<SearchbarInputEventDetail>,
+  ) => {
+    setSearchTerm(event.target.value);
+    if (event.detail.value) {
+      search(event.detail.value);
+    }
+  };
+
+  const handleSearchCancel = () => {
+    setSearchTerm(null);
+    setShowSearch(false);
+  };
+
+  const displaySamples = searchTerm ? searchResults || [] : samples;
+
+  return (
+    <>
+      <IonListHeader>
+        <IonLabel>Samples {samples && <>({samples.length})</>}</IonLabel>
+        <IonButton routerLink={paths.sampleCreate(submission.id)}>
+          New
+        </IonButton>
+      </IonListHeader>
+
+      <IonGrid className={showSearch ? "ion-hide" : ""}>
+        <IonRow class="ion-justify-content-between">
+          <IonCol size="auto">
+            <IonChip outline>
+              Template: {submission.metadata_submission.templates[0]}
+            </IonChip>
+          </IonCol>
+          <IonCol size="auto">
+            <IonButton
+              title="show sample search"
+              size="small"
+              fill="clear"
+              onClick={() => {
+                setShowSearch(true);
+                // Wait for the search bar to be rendered before focusing
+                setTimeout(() => searchElement.current?.setFocus(), 10);
```

Comment:
Here's a `useEffect` call that applies focus to the search field whenever the search field changes from hidden to visible. This could be used instead of the `setTimeout` call. This would go at the top level of the component (i.e. the same level as the `useState` calls).

```ts
  // Whenever the search field changes from hidden to visible, focus on it.
  useEffect(() => {
    if (showSearch) {
      searchElement.current?.setFocus();
    }
  }, [showSearch]);
```

Reply:
Yeah `autofocus` was the first thing I thought of, but Ionic recommends against it: https://ionicframework.com/docs/developing/managing-focus. The `useEffect` approach is interesting. I'll give that a shot.

Reply:
When I made the change to swap `setTimeout` for `useEffect` some of the tests in `SampleList.test.tsx` started failing. Not because `useEffect` is a bad idea, but I think because some of my previous test writing had been a bit sloppy. So this really sent me down a bit of a rabbit hole (a productive one!) of figuring out better ways to structure component tests. It also led me to a eslint plugin that enforces better practices around using `testing-library`.